src/utils/cache.rs

```diff
@@ -19,12 +19,22 @@ impl ServiceCache {
         let _: () = self.set_ex(id, dest, timeout).unwrap();
     }
 
+    pub fn invalidate_pattern(&self, pattern: &String) {
+        pipeline_delete(self, self.scan_match(pattern).unwrap());
+    }
+
     pub fn _invalidate(&self, id: &String) {
         let _: () = self.del(id).unwrap();
     }
 
-    pub fn cache_resp<R>(&self, key: &String, timeout: usize, resp: impl Fn() -> Result<R>) -> Result<content::Json<String>>
-        where R: Serialize
+    pub fn cache_resp<R>(
+        &self,
+        key: &String,
+        timeout: usize,
+        resp: impl Fn() -> Result<R>,
+    ) -> Result<content::Json<String>>
```

Comment:
Should we already return something serialized? Maybe we can make this a bit more "rocket" agnostic. Although, this beyond the scope of this PR but maybe it's better so somehow get rid of `content::Json<?>`